go.mod

```diff
@@ -129,7 +129,7 @@ require (
 )
 
 require (
-	cloud.google.com/go v0.97.0 // indirect
+	cloud.google.com/go/compute v1.7.0 // indirect
```

Comment:
Do we know if this will cause the publishing bot issues? I don't know how to tell in advance of merge

cc @dims @nikhita

Reply:
asked on slack - https://kubernetes.slack.com/archives/CHGFYJVAN/p1673321860957889?thread_ts=1672871786.359129&cid=CHGFYJVAN

Reply:
> Do we know if this will cause the publishing bot issues?

Yes, it will break `go mod tidy` due to https://github.com/golang/go/issues/27899. For example, here is what it'd look like while publishing apiserver:

```
   + go mod tidy
   go: downloading github.com/BurntSushi/toml v0.3.1
   go: downloading github.com/cockroachdb/datadriven v0.0.0-20200714090401-bf6692d28da5
   go: downloading cloud.google.com/go/compute v1.7.0
   go: downloading github.com/cockroachdb/errors v1.2.4
   go: downloading github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f
   go: downloading github.com/getsentry/raven-go v0.2.0
   go: downloading github.com/certifi/gocertifi v0.0.0-20200922220541-2c3bb06c6054
   go: downloading cloud.google.com/go v0.65.0
   k8s.io/apiserver/pkg/storage/storagebackend/factory imports
    go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc tested by
    go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.test imports
    google.golang.org/grpc/interop imports
    golang.org/x/oauth2/google imports
    cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
    cloud.google.com/go v0.65.0 (/home/nikhita/go/pkg/mod/cloud.google.com/go@v0.65.0/compute/metadata)
    cloud.google.com/go/compute v1.7.0 (/home/nikhita/go/pkg/mod/cloud.google.com/go/compute@v1.7.0/metadata)
```

It downloads `cloud.google.com/go@v0.65.0` because of:

```
go.opentelemetry.io/proto/otlp@v0.19.0 github.com/grpc-ecosystem/grpc-gateway/v2@v2.7.0
github.com/grpc-ecosystem/grpc-gateway/v2@v2.7.0 golang.org/x/oauth2@v0.0.0-20211104180415-d3ed0bb246c8
golang.org/x/oauth2@v0.0.0-20211104180415-d3ed0bb246c8 cloud.google.com/go@v0.65.0
```

One workaround for this is to replace `golang.org/x/oauth2` i.e. adding this line to all go.mod files:

```
replace golang.org/x/oauth2 => golang.org/x/oauth2 v0.0.0-20220722155238-128564f6959c
```

**Note**: If we go ahead with the workaround, it is important that _all_ changes to go.mod in this PR go in a single commit because the bot runs `go mod tidy` on each commit to be published.

Reply:
> One workaround for this is to replace `golang.org/x/oauth2` i.e. adding this line to all go.mod files:
> 
> ```
> replace golang.org/x/oauth2 => golang.org/x/oauth2 v0.0.0-20220722155238-128564f6959c
> ```

yeah, that's not an option... we can't use `replace` in modules intended to be used as libraries

it sounds like we have to wait for cloud.google.com/go deps to be cleaned up in https://github.com/kubernetes/kubernetes/issues/113366 before we can bump this :-/

Reply:
Thanks for the details. One idea -- can we switch back to use `cloud.google.com/go v0.97.0` in cAdvisor (i.e. partial revert of https://github.com/google/cadvisor/pull/3213/) to avoid this issue for now? 

I have a WIP PR - https://github.com/google/cadvisor/pull/3230, would it fix this issue?

Reply:
> I have a WIP PR - https://github.com/google/cadvisor/pull/3230, would it fix this issue? If so, I can merge that PR, and cut new release and re-vendor.

probably so? I find it hard to reason about the behavior, but you could always open a WIP k/k PR that pins to your fork / that branch to try it out

Reply:
Sounds good, I opened https://github.com/kubernetes/kubernetes/pull/114973 as a test

Reply:
The test worked. Cut a new cAdvisor release with that change and updated this PR to include it.